portfolio/src/main/webapp/script.js

```diff
@@ -29,3 +29,45 @@ function showSlides(n) {
   }
   slides[slideIndex-1].style.display = "block";
 }
+
+/**
+ * Fetches a message from the server and adds it to the page.
+ */
+function getMessage() {
+  console.log('Fetching the message');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addMsgToDom() function.
+  textPromise.then(addMsgToDom);
+}
```

Comment:
This is nice and readable. For better or worse I think the more idiomatic way to do this in javascript is more concisely stringing together arrow function expressions like:

```
fetch('/data').then(response => response.text()).then(message => addMsgToDom(message));
```

Reply:
that looks good, I changed it to this format